app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  BaseController,
+  RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+import type { DataDeletionService } from '../../services/data-deletion-service';
+import { DeleteRegulationStatus } from '../../../../shared/constants/metametrics';
+
+// Unique name for the controller
+const controllerName = 'MetaMetricsDataDeletionController';
+
+/**
+ * Get a type representing the public interface of the given type. The
+ * returned type will have all public properties, but will omit private
+ * properties.
+ *
+ * @template Interface - The interface to return a public representation of.
+ */
+type PublicInterface<Interface> = Pick<Interface, keyof Interface>;
```

Comment:
Now that `@metamask/utils` is updated to `>9.1.0` in develop, we could deduplicate the `PublicInterface` type here:

```suggestion
import {
  BaseController,
  RestrictedControllerMessenger,
} from '@metamask/base-controller';
import type { PublicInterface } from '@metamask/utils';
import type { DataDeletionService } from '../../services/data-deletion-service';
import { DeleteRegulationStatus } from '../../../../shared/constants/metametrics';

// Unique name for the controller
const controllerName = 'MetaMetricsDataDeletionController';
```